### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Shimwell**
This PR brings in the latest improvements to the examples and reduces the size of the docker image a little bit.

Hopefully this is enough to avoid running out of space while building the docker image on Github actions 

improvements to the tasks are the splitting up of DPA across isotopes in task 6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -189,18 +202,25 @@ ENV PATH=$PATH:/njoy/build
...
-    rm -rf nndc-b7.1-endf  && \
-    rm -rf nndc-b7.1-ace/  && \
-    rm -rf nndc-b7.1-download && \
+
+RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
+    mkdir nndc-b7.1-hdf5 && \
+    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
+    rm nndc-b7.1.tar.xz
+
+RUN git clone https://github.com/openmc-dev/data.git && \
```
Comment (Shimwell) [inline Dockerfile:219]:
```suggestion
RUN git clone --single-branch --branch master --depth 1 https://github.com/openmc-dev/data.git && \
```

**Comment by Shimwell** [comment]
Looks like I am getting the same DAGMC build error you mentioned @billingsley-john

File: Dockerfile [hunk 2]
```diff
@@ -178,9 +189,11 @@ RUN cd /opt && \
     pip install .
 
 #  NJOY2016 install from source
-RUN mkdir njoy && cd njoy && \
-    git clone --single-branch --branch master https://github.com/njoy/NJOY2016.git && \
-    mkdir build && cd build && \
+RUN mkdir njoy && \
+    cd njoy && \
+    git clone --single-branch --branch master --depth 1 https://github.com/njoy/NJOY2016.git && \
```
Comment (Shimwell) [inline Dockerfile:194]:
```suggestion
    git clone --single-branch --branch 2016.61 --depth 1 https://github.com/njoy/NJOY2016.git && \
```

File: Dockerfile [hunk 3]
```diff
@@ -189,18 +202,25 @@ ENV PATH=$PATH:/njoy/build
...
-    rm -rf nndc-b7.1-ace/  && \
-    rm -rf nndc-b7.1-download && \
+
+RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
+    mkdir nndc-b7.1-hdf5 && \
+    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
+    rm nndc-b7.1.tar.xz
+
+RUN git clone --single-branch --branch master --depth 1 https://github.com/openmc-dev/data.git && \
+    # python data/convert_nndc71.py --cleanup && \
```
Comment (Shimwell) [inline Dockerfile:220]:
```suggestion
    pip install neutronics-material-maker && \
    openmc_data_downloader -l ENDFB-7.1-NNDC -p photon -e H He Li Be B C N O F Ne Na Mg Al Si P S Cl Ar K Ca Sc Ti V Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr Rb Sr Y Zr Nb Mo Ru Rh Pd Ag Cd In Sn Sb Te I Xe Cs Ba La Ce Pr Nd Sm Eu Gd Tb Dy Ho Er Tm Yb Lu Hf Ta W Re Os Ir Pt Au Hg Tl Pb Bi Th Pa U 
    # python data/convert_nndc71.py --cleanup && \
```

File: Dockerfile [hunk 4]
```diff
@@ -189,18 +202,25 @@ ENV PATH=$PATH:/njoy/build
...
-RUN git clone https://github.com/openmc-dev/data.git
-RUN python data/convert_nndc71.py --cleanup && \
-    rm -rf nndc-b7.1-endf  && \
-    rm -rf nndc-b7.1-ace/  && \
-    rm -rf nndc-b7.1-download && \
+
+RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
+    mkdir nndc-b7.1-hdf5 && \
+    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
+    rm nndc-b7.1.tar.xz
```
Comment (Shimwell) [inline Dockerfile:217]:
```suggestion
# RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
#    mkdir nndc-b7.1-hdf5 && \
#    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
#    rm nndc-b7.1.tar.xz
```

File: Dockerfile [hunk 5]
```diff
@@ -178,52 +189,66 @@ RUN cd /opt && \
...
-    python data/combine_libraries.py -l nndc-b7.1-hdf5/cross_sections.xml tendl-2019-hdf5/cross_sections.xml -o cross_sections.xml && \
-    python delete_nuclear_data_not_used_in_cross_section_xml.py
 
+# RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
+#    mkdir nndc-b7.1-hdf5 && \
+#    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
+#    rm nndc-b7.1.tar.xz
+
+RUN git clone --single-branch --branch master --depth 1 https://github.com/openmc-dev/data.git && \
+    pip install neutronics-material-maker && \
```
Comment (Shimwell) [inline Dockerfile:221]:
```suggestion
    pip install openmc_data_downloader && \
```

**Comment by Shimwell** [comment]
Looks like the nuclear data downloader is working and the amount of space required for the nuclear data now fits into the docker image.

Just need to update all the tasks to check they work with the new nuclear data including task 1 which is plotting cross sections

File: Dockerfile [hunk 6]
```diff
@@ -178,52 +189,66 @@ RUN cd /opt && \
...
-    python delete_nuclear_data_not_used_in_cross_section_xml.py
 
+# RUN wget -O nndc-b7.1.tar.xz https://anl.box.com/shared/static/9igk353zpy8fn9ttvtrqgzvw1vtejoz6.xz && \
+#    mkdir nndc-b7.1-hdf5 && \
+#    tar -xf nndc-b7.1.tar.xz -C nndc-b7.1-hdf5 && \
+#    rm nndc-b7.1.tar.xz
+
+RUN git clone --single-branch --branch master --depth 1 https://github.com/openmc-dev/data.git && \
+    pip install openmc_data_downloader && \
+    openmc_data_downloader -l ENDFB-7.1-NNDC TENDL-2019 -d cross_section_data -e H He Li Be B C N O F Ne Na Mg Al Si P S Cl Ar K Ca Sc Ti V Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr Rb Sr Y Zr Nb Mo Ru Rh Pd Ag Cd In Sn Sb Te I Xe Cs Ba La Ce Pr Nd Sm Eu Gd Tb Dy Ho Er Tm Yb Lu Hf Ta W Re Os Ir Pt Au Hg Tl Pb Bi Th Pa U
```
Comment (Shimwell) [inline Dockerfile:222]:
```suggestion
    openmc_data_downloader -l ENDFB-7.1-NNDC TENDL-2019 -d cross_section_data -p neutron photon -e H He Li Be B C N O F Ne Na Mg Al Si P S Cl Ar K Ca Sc Ti V Cr Mn Fe Co Ni Cu Zn Ga Ge As Se Br Kr Rb Sr Y Zr Nb Mo Ru Rh Pd Ag Cd In Sn Sb Te I Xe Cs Ba La Ce Pr Nd Sm Eu Gd Tb Dy Ho Er Tm Yb Lu Hf Ta W Re Os Ir Pt Au Hg Tl Pb Bi Th Pa U
```

**Review by billingsley-john — APPROVED** [review]
Looks good to go when task 1 has been updated

**Pull request merged**